server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/VacationRelayIntegrationTest.java

```diff
@@ -0,0 +1,192 @@
+
+
+    @Rule
+    public final SwarmGenericContainer fakeSmtp = new SwarmGenericContainer("weave/rest-smtp-sink:latest");
+
+    private static final String DOMAIN = "mydomain.tld";
+    private static final String USER = "benwa";
+    private static final String USER_WITH_DOMAIN = USER + '@' + DOMAIN;
+    private static final String PASSWORD = "secret";
+    private static final String REASON = "Message explaining my wonderful vacations";
+
+    private static final String LOCALHOST_IP = "127.0.0.1";
+    private static final int SMTP_PORT = 1025;
+
+    private ConditionFactory calmlyAwait;
+    private GuiceJamesServer guiceJamesServer;
+    private JmapGuiceProbe jmapGuiceProbe;
+
+    private RequestSpecification jmapRequestSpecification;
+    private RequestSpecification restSmtpSinkSpec;
+
+    protected abstract void await();
+
+    protected abstract GuiceJamesServer getJmapServer();
+
+    protected abstract InMemoryDNSService getInMemoryDns();
+
+    @Before
+    public void setUp() throws Exception {
+
+        InetAddress containerIp = InetAddresses.forString(fakeSmtp.getIp());
+        getInMemoryDns()
+            .registerRecord("yopmail.com", containerIp, "yopmail.com");
+
+        guiceJamesServer = getJmapServer();
+        guiceJamesServer.start();
+
+        DataProbe dataProbe = guiceJamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DOMAIN);
+        dataProbe.addUser(USER_WITH_DOMAIN, PASSWORD);
+        MailboxProbe mailboxProbe = guiceJamesServer.getProbe(MailboxProbeImpl.class);
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, USER_WITH_DOMAIN, DefaultMailboxes.SENT);
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, USER_WITH_DOMAIN, DefaultMailboxes.INBOX);
+        await();
+
+        jmapGuiceProbe = guiceJamesServer.getProbe(JmapGuiceProbe.class);
+        jmapRequestSpecification = new RequestSpecBuilder()
+        		.setContentType(ContentType.JSON)
+        		.setAccept(ContentType.JSON)
+        		.setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8)))
+        		.setPort(jmapGuiceProbe
+                    .getJmapPort())
+        		.build();
+
+        restSmtpSinkSpec = new RequestSpecBuilder()
+            .setContentType(ContentType.JSON)
+            .setAccept(ContentType.JSON)
+            .setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8)))
+            .setPort(80)
+            .setBaseUri("http://" + containerIp.getHostAddress())
+            .build();
+
+        Duration slowPacedPollInterval = Duration.FIVE_HUNDRED_MILLISECONDS;
+        calmlyAwait = Awaitility.with().pollInterval(slowPacedPollInterval).and().with().pollDelay(slowPacedPollInterval).await();
+    }
+
+    @After
+    public void teardown() {
+        guiceJamesServer.stop();
+    }
+
+    private void setVacationResponse(AccessToken accessToken) {
```

Comment:
You can do this via the JMAP guice probe....

You don't need at all to interact with JMAP APIs